pkg/scheduler/plugins/noderesourcefitplus/node_resources_fit_plus.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcesfitplus
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/api/v1/resource"
+	k8sConfig "k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	plfeature "k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/noderesources"
+
+	"github.com/koordinator-sh/koordinator/pkg/scheduler/apis/config"
+)
+
+const (
+	// Name is plugin name
+	Name = "NodeResourcesFitPlus"
+)
+
+var (
+	_ framework.ScorePlugin = &Plugin{}
+)
+
+type Plugin struct {
+	handle framework.Handle
+	args   *config.NodeResourcesFitPlusArgs
+}
+
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+
+	sampleArgs2, ok := args.(*config.NodeResourcesFitPlusArgs)
+
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesArgs, got %T", args)
+	}
+
+	return &Plugin{
+		handle: handle,
+		args:   sampleArgs2,
+	}, nil
+}
+
+func (s *Plugin) Name() string {
+	return Name
+}
+
+func (s *Plugin) Score(ctx context.Context, state *framework.CycleState, p *v1.Pod, nodeName string) (int64, *framework.Status) {
+
+	nodeInfo, err := s.handle.SnapshotSharedLister().NodeInfos().Get(nodeName)
+	if err != nil {
+		return 0, framework.NewStatus(framework.Error, fmt.Sprintf("getting node %q from Snapshot: %v", nodeName, err))
+	}
+
+	var nodeScore int64
+	var weightSum int64
+
+	podRequest, _ := fitsRequest(computePodResourceRequest(p).Resource, nodeInfo)
```

Comment:
why not cumpute it in PreScore and save to preScoreState

Reply:
The starting point of this problem is to store the output results of fitsRequest into preScoreState through PreScore, and then let the noderesourcesfitplus and scarceresourceavoidance plug-ins share data?

Reply:
@ZiMengSheng done